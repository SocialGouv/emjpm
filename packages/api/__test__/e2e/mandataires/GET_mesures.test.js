//

const request = require("supertest");
const server = require("@emjpm/api/app");
const knex = require("@emjpm/api/db/knex");

const { getTokenByUserType } = require("../utils");

beforeAll(async () => {
  await knex.seed.run();
});

afterAll(async () => {
  await knex.destroy();
});

test("should get list of mesures", async () => {
  const token = await getTokenByUserType("mandataire");
  const response = await request(server)
    .get("/api/v1/mandataires/1/mesures")
    .set("Authorization", "Bearer " + token);

  expect(response.body).toMatchSnapshot();
  expect(response.status).toBe(200);
});

test("should get list of mesures for maps", async () => {
  const token = await getTokenByUserType("mandataire");
  const response = await request(server)
    .get("/api/v1/mandataires/1/mesuresForMaps")
    .set("Authorization", "Bearer " + token);

  expect(response.body).toMatchSnapshot();
  expect(response.status).toBe(200);
});

test("should get list of mesures en attente", async () => {
  const token = await getTokenByUserType("mandataire");
  const response = await request(server)
    .get("/api/v1/mandataires/1/mesures/attente")
    .set("Authorization", "Bearer " + token);

  expect(response.body).toMatchSnapshot();
  expect(response.status).toBe(200);
});

test("should get list of mesures eteintes", async () => {
  const token = await getTokenByUserType("mandataire");
  const response = await request(server)
    .get("/api/v1/mandataires/1/mesures/Eteinte")
    .set("Authorization", "Bearer " + token);

  expect(response.body).toMatchSnapshot();
  expect(response.status).toBe(200);
});
