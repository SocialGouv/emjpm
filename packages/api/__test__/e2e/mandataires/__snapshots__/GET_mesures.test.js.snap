// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get list of mesures 1`] = `
Array [
  Object {
    "annee": "2010-10-05",
    "cabinet": null,
    "civilite": "F",
    "code_postal": "62000",
    "created_at": "2010-10-05T00:00:00.000Z",
    "date_ouverture": "2010-10-04T22:00:00.000Z",
    "etablissement": "UDAHF",
    "etablissement_id": null,
    "extinction": null,
    "id": 1,
    "mandataire_id": 1,
    "numero_dossier": null,
    "residence": "oui",
    "status": "Mesure en cours",
    "ti_id": null,
    "type": "preposes",
    "ville": "Arras",
  },
  Object {
    "annee": "2010-10-05",
    "cabinet": null,
    "civilite": "H",
    "code_postal": "93000",
    "created_at": "2010-10-05T00:00:00.000Z",
    "date_ouverture": "2010-10-04T22:00:00.000Z",
    "etablissement": "CAROUF",
    "etablissement_id": null,
    "extinction": null,
    "id": 4,
    "mandataire_id": 1,
    "numero_dossier": null,
    "residence": "oui",
    "status": "Mesure en cours",
    "ti_id": null,
    "type": "preposes",
    "ville": "St Denis",
  },
]
`;

exports[`should get list of mesures en attente 1`] = `
Array [
  Object {
    "annee": "2010-10-05",
    "cabinet": null,
    "civilite": "F",
    "code_postal": "93000",
    "created_at": "2010-10-05T00:00:00.000Z",
    "date_ouverture": "2010-10-04T22:00:00.000Z",
    "etablissement": null,
    "etablissement_id": null,
    "extinction": null,
    "id": 3,
    "mandataire_id": 1,
    "numero_dossier": null,
    "residence": null,
    "status": "Mesure en attente",
    "ti_id": null,
    "type": "preposes",
    "ville": null,
  },
]
`;

exports[`should get list of mesures eteintes 1`] = `
Array [
  Object {
    "annee": "2012-10-05",
    "cabinet": null,
    "civilite": "F",
    "code_postal": "92000",
    "created_at": "2012-10-05T00:00:00.000Z",
    "date_ouverture": "2012-10-04T22:00:00.000Z",
    "etablissement": "CRF1",
    "etablissement_id": null,
    "extinction": null,
    "id": 5,
    "mandataire_id": 1,
    "numero_dossier": null,
    "residence": "oui",
    "status": "Eteindre mesure",
    "ti_id": null,
    "type": "preposes",
    "ville": "St ambroise",
  },
]
`;

exports[`should get list of mesures for maps 1`] = `
Array [
  Object {
    "array_agg": Array [
      "preposes 2010-10-05",
    ],
    "code_postal": "93000",
    "count": "1",
    "latitude": 0.3,
    "longitude": 0.3,
  },
  Object {
    "array_agg": Array [
      "preposes 2010-10-05",
    ],
    "code_postal": "62000",
    "count": "1",
    "latitude": 50.5333,
    "longitude": 2.6333,
  },
]
`;
